### ISSUE TITLE
Incorrect texture used

### ISSUE THREAD

**Issue by hiker** [supertuxkart/stk-code#1891]
The addons kart 'alternative tux' uses a wheel texture 'wheel.png'. Since it does not declare this texture in its materials.xml file, it is not loaded, instead the wheel.png texture from 20_amanda is used (which in turn causes warnings, since the amanda texture uses a gloss map, which is then not found).

**Comment by hiker** [comment]
I've marked this as major since stk actually crashes with the incorrect texture when selecting the addons tab (with only Alternative Tux installed, and not fixing the texture).

Adding the 'wheel.png' texture to Alt.Tux's materials.xml file fixes the issue. So one solution would be to modify the exporter to list all textures used - though then it's difficult to decide what to do with the ones that are shared :(

**Comment by auriamg** [comment]
This happens because karts use shared materials

```
    // addShared makes sure that these textures/material infos stay in memory
    material_manager->addSharedMaterial(materials_file);
```

Possibly we need to figure a way to make shared materials have something like "scopes"? For instance add a shared material that is scoped to be applicable only for "kart alternative_tux"

**Comment by math07** [comment]
Great idea!

**Comment by auriamg** [comment]
Possible fix pushed in 81bd79a73662190b6cacf509e0440eb276eb3d2e

**Comment by auriamg** [comment]
Confirmed fixed with c791fd46c13f1147b6cbbb91641aad348c63ca55
